### ISSUE TITLE
unable to compile with Alpine Linux

### ISSUE THREAD

**Issue by k4gi** [crawl/crawl#2446]
**PLATFORM and OS or BROWSER:**
console version on postmarketos v21.12.2 on pinephone

**DCSS VERSION AND GAME INFORMATION:**
0.28.0

**DESCRIPTION:**
I attempted compiling a few times until I had sorted out installing all the dependencies. When it got to the file 'crash.o' there is an error:

`crash.cc:47:10: fatal error: execinfo.h: No such file or directory`

for whatever enlightened reason Alpine, and therefore PostmarketOS, does not (seem to) have execinfo.h or a drop-in replacement.

I can install the package `libexecinfo` which is a dummy and leads to complaints of undefined references to `backtrace`, `backtrace_symbols`, and `backtrace_symbols_fd`

There is another package `libexecinfo-dev` idk what thats about, didnt seem to make any difference

apparently there is a package `libunwind` that is a replacement, but individual projects need to support using it.

**EXTRA FILES:**
Here is an issue in another project I found having the same problem. node.js stuff goes right over my head...

https://github.com/nodejs/node-report/issues/86

**Comment by rawlins** [comment]
This is a known issue but I'm not sure there's a currently open report for it. To just get crawl to build is doable; There's a PR that never got finished enough to merge, but does have some code that tries to work around this: https://github.com/crawl/crawl/pull/1151. Even more simply, you can `#undef BACKTRACE_SUPPORTED` and it should build.

Sort of unclear to me what the general solution is (at least without autoconf-style build infrastructure, which we don't use).

**Comment by jv84** [comment]
Another distro specific issue. Requesting to close this.

**Comment by jv84** [comment]
There's no libexecinfo package since Alpine 3.17. Why support compilation?

**Comment by erikmack** [comment]
I maintain the crawl package for Gentoo Linux, which supports some alternative libc's like musl and dietlibc.  libexecinfo is specific to glibc (GNU libc) so a system with any other libc will fail to build.  Come to think of it, FreeBSD has crawl so maybe their libc has libexecinfo also.  Alternative libc's are usually designed to be lightweight for very constrained systems.

All this is to say that the issue isn't specific to Alpine or any distro.

This is the issue reported by Gentoo's CI: https://bugs.gentoo.org/831632 and here's the tracking ticket for other packages that fail the same way: https://bugs.gentoo.org/856418

Next time I push an update I'll try the crash.cc patch mentioned above and report back.

**Comment by rawlins** [comment]
I don't think this needs to be closed; if it's possible and not too involved, we would be willing to support alternative libcs on linux (and for that matter, already have infrastructure for handling this for mac/windows).

**Comment by draeath** [comment]
#1151 was closed for no good reason without 17844727410be6fc8dd6258b0868848569472a4f being accepted.

Linking directly to it so folks can at least patch it locally - but it would be nice for this to be handled properly.
